core/trino-main/src/main/java/io/trino/sql/planner/DeterminismEvaluator.java

```diff
@@ -63,7 +63,7 @@ protected Void visitFunctionCall(FunctionCall node, AtomicBoolean deterministic)
             if (!resolvedFunctionSupplier.apply(node).isDeterministic()) {
                 deterministic.set(false);
             }
-            return super.visitFunctionCall(node, deterministic);
```

Comment:
This is necessary so that the arguments to the function are visited.

Reply:
Exactly, I put the wrong place